### DISCUSSION TITLE
Thin clients writing changes back to server

### DISCUSSION THREAD

**Discussion by MacMilanzi** [ltsp/ltsp#685] [Q&A]
Hi,
I am running the LTSP on Ubuntu 20.04 LTS and it has been amazing so far but i have a question. When i boot my thin client and make a change on the thin client it writes that change back to the server. Is there a setting to control that?

**Comment by alkisg** [comment]
Hi, what kind of change?
E.g. do you `apt install a-program` or do you `create a file under /home/username`?

**Reply by MacMilanzi** [reply]
I create a file under the /home/username and sometimes I can do apt install
if I don’t require sudo privileges.

**Comment by alkisg** [comment]
When you create files in /home/username of course they're expected to show up in the server.
Are you asking "how to create guest accounts"?
I don't understand the problem, please describe it in more detail. What are you doing, what are you expecting to see, and what do you actually see.

**Reply by MacMilanzi** [reply]
Ok, so here is what am trying to do. The thin clients are out at some offices but i don't want the user to make changes to the systems settings on them that affect the initial image, i want to be able to reboot the thin clients to the default image if this happens. for example, if they have installed lets say a program and that program is misbehaving, I want to be to just tell them to reboot and the thin client grabs the default image which should not have that misbehaving program they installed.

Right now if i make any systems setting change on the thin clients the changes stick even if i reboot the thin clients. does that make sense?

**Comment by alkisg** [comment]
Are you using thin clients or fat clients? If you open a terminal, does it show `user@servername` or `user@ltsp123`?
By default, the new LTSP is using fat clients. This means that changes to /home/username persist reboots, but outside it they don't.
I.e. if a user runs `sudo apt install program`, the installed program is lost after reboot.

**Reply by MacMilanzi** [reply]
On the terminal I am seeing user@ltsp123, so does that mean am running fat clients?  if yes, how do i use thin clients instead of fat clients so that the changes do not persist after reboot? Can I be able to switch back and forth fat to thin clients as well?

**Comment by alkisg** [comment]
In fat clients, changes do not persist reboots. Try to create a file in the root directory, then reboot, and see that it's not there:

```shell
user@ltsp123:~$ sudo touch /test
user@ltsp123:~$ reboot
```

And then after rebooting:

```shell
user@ltsp123:~$ ls /test
ls: cannot access '/test': No such file or directory
```

**Reply by MacMilanzi** [reply]
Okay i double checked, I am running thin clients. My mistake above i said it was showing user@ltsp123, No its not. the terminal is showing the user@servername, and i have followed your test above and the directory does exist after reboot. so the change persists.

**Comment by alkisg** [comment]
Which ltsp version are you running? What's the output of `dpkg -l | grep ltsp`?

**Reply by MacMilanzi** [reply]
It shows:
ii ltsp 20.04-1  all  Linux Terminal Server Project

**Comment by alkisg** [comment]
Then you're reporting something wrong, as LTSP 20.04-1 doesn't support thin clients at all.
You need to describe your setup in more detail.
E.g. if you're using x2go or VNC to connect to the server, it's not about LTSP anymore, it's about x2go and VNC...

**Comment by MacMilanzi** [comment]
Okay, yes.. yes you are right. something was wrong with my vm sorry. Here is what i have double checked based on our conversation.
1. The clients terminals are showing user@ltsp123
2. After running user@ltsp123:~$ sudo touch /test
user@ltsp123:~$ reboot
I got:  user@ltsp123:~$ ls /test
ls: cannot access '/test': No such file or directory

I am running the LTSP server on Ubuntu 20.04 on DELL PowerEdge 440 VMware Esxi as a VM. For the clients I have the OnLogic thin client that i boot to grab the image.

**Comment by MacMilanzi** [comment]
But then this confusing me. this test shows that the changes do not persist but the when i do system setting change it persist. Is this normal?

**Comment by alkisg** [comment]
I wrote very specific commands to test persistence, and we still needed to exchange a lot of messages in order for you to verify them.

So when you say "system changes persist" noone will be able to help you because it's too vague.
Describe specific, reproducible steps so that people can understand what you're trying to do. Even if you have to say "I click on this or the other button" and upload screenshots.

**Comment by MacMilanzi** [comment]
Okay: on the user /home/user/
I have a script that starts up on boot. /home/user/connection.conf
Its executable, when this script is run it connects to a remote desktop using xfreerdp

on the client:
I go to settings --> applications --> startup applications --> untick this startup script setting. 
(However on the server, that user startup application setting is still intact and ticked to run on startup.)
then i go to the terminal and reboot.
user@ltsp123:~$ reboot

After reboot the client, the script does not run on startup.

Next:
I change the background image on the client, but not on the server for that user of that client. I reboot, the background remains.

**Comment by alkisg** [comment]
This is a user change, not a system change. User changes are saved in /home/username/.config.
The /home/username is mounted via SSHFS from the LTSP server, so the user changes are expected to persist. Otherwise you wouldn't be able to save documents, firefox bookmarks etc.

**Comment by MacMilanzi** [comment]
Okay. Is there a way to limit what the user can change then?

**Comment by alkisg** [comment]
Describe again with simple steps what limit you want to impose on your users.
If you want to create a user process that the users can't block or kill, it should probably NOT be a user process but a system process.
If you want to create a user setting that user's can't override, check your desktop environment manual for "mandatory settings".

Both of these tasks aren't really related to LTSP, so they should be directed to distribution communities instead...

**Comment by MacMilanzi** [comment]
Alright. Thank you. I will do that then. I thought they would be related because i have scripts that i want to run on startup of the clients and that cant be killed by the user. I though there might be a way like to configure the scripts to run on startup in the ltsp.conf file.

But thank you for your clarification.
